ctapipe/io/dl1eventsource.py

```diff
@@ -0,0 +1,350 @@
+                            PeakTimeStatisticsContainer(),
+                        ],
+                        prefixes=True,
+                    )
+                    for tel in self.file_.root.dl1.event.telescope.parameters
+                }
+
+        if self.is_simulation:
+            # true shower wide information
+            mc_shower_reader = HDF5TableReader(self.file_).read(
+                "/simulation/event/subarray/shower", MCEventContainer(), prefixes="true"
+            )
+
+        # Setup iterators for the array events
+        events = HDF5TableReader(self.file_).read(
+            "/dl1/event/subarray/trigger", [TriggerContainer(), EventIndexContainer()]
+        )
+        for counter, array_event in enumerate(events):
+            data.dl1.tel.clear()
+            data.mc.tel.clear()
+            data.pointing.tel.clear()
+            data.trigger.tel.clear()
+
+            data.count = counter
+            data.trigger, data.index = next(events)
+            data.trigger.tels_with_trigger = (
+                np.where(data.trigger.tels_with_trigger)[0] + 1
+            )  # +1 to match array index to telescope id
+
+            # Maybe there is a simpler way  to do this
+            # Beware: tels_with_trigger contains all triggered telescopes whereas
+            # the telescope trigger table contains only the subset of
+            # allowed_tels given during the creation of the dl1 file
+            for i in self.file_.root.dl1.event.telescope.trigger.where(
+                f"(obs_id=={data.index.obs_id}) & (event_id=={data.index.event_id})"
+            ):
+                if self.allowed_tels and i["tel_id"] not in self.allowed_tels:
+                    continue
+                data.trigger.tel[i["tel_id"]].time = i["telescopetrigger_time"]
+
+            self._fill_array_pointing(data)
+            self._fill_telescope_pointing(data)
+
+            if self.is_simulation:
+                data.mc = next(mc_shower_reader)
+                data.mcheader = self._mc_headers[data.index.obs_id]
+
+            for tel in data.trigger.tel.keys():
+                if self.allowed_tels and tel not in self.allowed_tels:
+                    continue
+                dl1 = data.dl1.tel[tel]
+                if DataLevel.DL1_IMAGES in self.datalevels:
+                    if f"tel_{tel:03d}" not in image_iterators.keys():
+                        print("miss", tel)
+                        continue
+                    image_row = next(image_iterators[f"tel_{tel:03d}"])
+                    dl1.image = image_row["image"]
+                    dl1.peak_time = image_row["peak_time"]
+                    dl1.image_mask = image_row["image_mask"]
+                    # ToDo: Find a place in the data container, where the true images can go #1368
+
+                if DataLevel.DL1_PARAMETERS in self.datalevels:
+                    if f"tel_{tel:03d}" not in param_readers.keys():
+                        print("miss", tel)
```

Comment:
print statement

Reply:
Should be logging or raising a warning

Reply:
Leftovers from debugging. This affects the case where the dl1 file was constructed with only a subset of telescopes (but the trigger table still contains all triggered telescopes).
We don't really log much in the event sources, but it probably doesn't hurt to start somewhere.